stdlib/public/core/Filter.swift.gyb

```diff
@@ -236,21 +236,35 @@ public struct ${Self}<
 %   if Traversal == 'Bidirectional':
   @warn_unused_result
   public func index(before i: Index) -> Index {
-    fatalError("FIXME: swift-3-indexing-model: implement")
+    return LazyFilterIndex(base: _previousFiltered(i.base))
   }
 %   end
 
   // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
   public func index(_ i: Index, offsetBy n: IndexDistance) -> Index {
```

Comment:
Ooh, that makes sense. I'll update the PR after ensuring everything still builds.